cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -415,7 +427,9 @@ public ApplicationDetail getLatestAppDetail(ApplicationReference appRef)
    *                                                reference is not found.
    */
   public SourceControlMetadataRecord getSourceControlMetadataRecord(ApplicationReference appRef)
-      throws SourceControlMetadataNotFoundException {
+      throws SourceControlMetadataNotFoundException, RepositoryNotFoundException {
+    // Checking if repository config exists for the namespace
```

Comment:
IIRC RepositoryNotFoundException would be interpreted as 404 right

Reply:
Yes, right